contracts/Overtime/FreeBetsHolder.sol

```diff
@@ -0,0 +1,164 @@
+        setOwner(_owner);
+        sportsAMM = ISportsAMMV2(_sportsAMMV2);
+        liveTradingProcessor = ILiveTradingProcessor(_liveTradingProcessor);
+    }
+
+    function fund(address _user, address _collateral, uint _amount) external notPaused {
+        require(supportedCollateral[_collateral], "Unsupported collateral");
+        IERC20(_collateral).safeTransferFrom(msg.sender, address(this), _amount);
+        balancePerUserAndCollateral[_user][_collateral] += _amount;
+        emit UserFunded(_user, _collateral, _amount, msg.sender);
+    }
+
+    function trade(
+        ISportsAMMV2.TradeData[] calldata _tradeData,
+        uint _buyInAmount,
+        uint _expectedPayout,
+        uint _additionalSlippage,
+        address _referrer,
+        address _collateral
+    ) external notPaused canTrade(msg.sender, _collateral, _buyInAmount) {
+        balancePerUserAndCollateral[msg.sender][_collateral] -= _buyInAmount;
+        address createdTicket = sportsAMM.trade(
+            _tradeData,
+            _buyInAmount,
+            _expectedPayout,
+            _additionalSlippage,
+            address(0),
+            _referrer,
+            _collateral == address(sportsAMM.defaultCollateral()) ? address(0) : _collateral,
+            false
+        );
+        ticketToUser[createdTicket] = msg.sender;
+    }
+
+    function tradeLive(
+        bytes32 _gameId,
+        uint16 _sportId,
+        uint16 _typeId,
+        uint8 _position,
+        uint _buyInAmount,
+        uint _expectedPayout,
+        uint _additionalSlippage,
+        address _referrer,
+        address _collateral
+    ) external notPaused canTrade(msg.sender, _collateral, _buyInAmount) {
+        bytes32 _requestId = liveTradingProcessor.requestLiveTrade(
+            _gameId,
+            _sportId,
+            _typeId,
+            _position,
+            _buyInAmount,
+            _expectedPayout,
+            _additionalSlippage,
+            address(0),
+            _referrer,
+            _collateral
+        );
+        liveRequestsPerUser[_requestId] = msg.sender;
+    }
+
+    function confirmLiveTrade(
+        bytes32 requestId,
+        address _createdTicket,
+        uint _buyInAmount,
+        address _collateral
+    ) external notPaused {
+        address _user = liveRequestsPerUser[requestId];
+        require(_user != address(0), "Unknown live ticket");
+
+        if (_collateral == address(0)) {
+            _collateral = address(sportsAMM.defaultCollateral());
+        }
+
+        require(supportedCollateral[_collateral], "Unsupported collateral");
+        require(balancePerUserAndCollateral[_user][_collateral] >= _buyInAmount, "Insufficient balance");
+
+        balancePerUserAndCollateral[_user][_collateral] -= _buyInAmount;
+        ticketToUser[_createdTicket] = _user;
+    }
+
+    function claimTicket(address _ticket) external notPaused {
```

Comment:
I would put nonReentrancy here mainly because there are funds claimed

Reply:
claim ticket is no longer needed as its all done via exercize workflow